activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -110,6 +110,30 @@ def index_by
     end
   end
 
+  # Finds the first element matching the specified attributes.
+  #
+  #   people = [
+  #     Person.new(first_name: "Jane", last_name: "Doe", age: 23),
+  #     Person.new(first_name: "John", last_name: "Doe", age: 42),
+  #     Person.new(first_name: "John", last_name: "Doe", age: 32)
+  #   ]
+  #
+  #   people.find_by(first_name: 'John', last_name: 'Doe')
+  #   # => #<Person first_name: "John", last_name: "Doe", age: 42>
+  #
+  #   people.find_by(last_name: 'Doe')
+  #   # => #<Person first_name: "Jane", last_name: "Doe", age: 23>
+  #
+  #   people.find_by(age: 32)
+  #   # => #<Person first_name: "John", last_name: "Doe", age: 32>
+  def find_by(attrs)
```

Comment:
We need to document in the core extensions guide as well. https://github.com/rails/rails/blob/main/guides/source/active_support_core_extensions.md

Reply:
Added. Let me know if something else is needed. 😃